src/libtomahawk/utils/ResultUrlChecker.h

```diff
@@ -33,10 +33,11 @@ class ResultUrlChecker : public QObject
 {
     Q_OBJECT
 public:
-    ResultUrlChecker( const query_ptr& query, const QList< result_ptr >& results );
+    ResultUrlChecker( const query_ptr& query, QObject* userData, const QList< result_ptr >& results );
     virtual ~ResultUrlChecker();
 
     query_ptr query() const { return m_query; }
+    void* userData() const { return m_userData; }
```

Comment:
This should probably also return a QObject* then. Ideally even a QPointer to a QObject, so we can don't accidentally access a deleted object. This also enables us to use a qobject_case instead of reinterpreting, which should be faster.